sycamore/llms/openai.py

```diff
@@ -0,0 +1,227 @@
+            return OpenAIClient(
+                api_key=self.api_key,
+                organization=self.organization,
+                base_url=self.base_url,
+                max_retries=self.max_retries,
+                **self.extra_kwargs,
+            )
+        elif self.client_type == OpenAIClientType.AZURE:
+            if self.azure_endpoint is None:
+                raise ValueError("azure_endpoint must be specified for Azure clients.")
+
+            return AzureOpenAIClient(
+                azure_endpoint=self.azure_endpoint,
+                azure_deployment=self.azure_deployment,
+                api_version=self.api_version,
+                api_key=self.api_key,
+                azure_ad_token=self.azure_ad_token,
+                azure_ad_token_provider=self.azure_ad_token_provider,
+                organization=self.organization,
+                max_retries=self.max_retries,
+                **self.extra_kwargs,
+            )
+
+        else:
+            raise ValueError(f"Invalid client_type {self.client_type}")
+
+    def get_guidance_model(self, model) -> Model:
+        if self.client_type == OpenAIClientType.OPENAI:
+            return GuidanceOpenAI(
+                model=model,
+                api_key=self.api_key,
+                organization=self.organization,
+                base_url=self.base_url,
+                max_retries=self.max_retries,
+                **self.extra_kwargs,
+            )
+        elif self.client_type == OpenAIClientType.AZURE:
+            return GuidanceAzureOpenAI(
+                model=model,
+                azure_endpoint=self.azure_endpoint,
+                azure_ad_token_provider=self.azure_ad_token_provider,
+                api_key=self.api_key,
+                version=self.api_version,
+                azure_deployment=self.azure_deployment,
+                azure_ad_token=self.azure_ad_token,
+                organization=self.organization,
+                max_retries=self.max_retries,
+                **self.extra_kwargs,
+            )
+
+        else:
+            raise ValueError(f"Invalid client_type {self.client_type}")
+
+
+# Allow rough backwards compatibility
+OpenAIClientParameters = OpenAIClientWrapper
+
+
+class OpenAI(LLM):
+    def __init__(
+        self,
+        model_name: Union[OpenAIModels, OpenAIModel, str],
+        api_key=None,
+        client_wrapper: Optional[OpenAIClientWrapper] = None,
+        params: Optional[OpenAIClientParameters] = None,
+        **kwargs,
+    ):
+        if isinstance(model_name, OpenAIModels):
+            self.model = model_name.value
+        elif isinstance(model_name, OpenAIModel):
+            self.model = model_name
+        elif isinstance(model_name, str):
+            self.model = OpenAIModels.from_name(model_name).value
+
+        if self.model.name == OpenAIModels.TEXT_DAVINCI.value.name:
+            print("text-davinci-003 is deprecated. Falling back to gpt-3.5-turbo-instruct")
```

Comment:
side note: we should probably use a logger everywhere

Reply:
Sure, will update.